backend/src/main/java/capstone/backend/services/ProductService.java

```diff
@@ -36,26 +40,32 @@ public ProductDTO getProductDetails(Long id) {
     }
 
     public ProductDTO createProduct(ProductDTO product) throws EntityWithThisIdAlreadyExistException {
-        if (product.getId() != null && repo
-                .findById(product.getId())
-                .isPresent()) {
+        if (productExists(product)) {
             throw new EntityWithThisIdAlreadyExistException(String.format("Product %s already has the id %d", product.getName(), product.getId()));
         }
         return ProductMapper.mapProductWithDetails(repo.
                 save(ProductMapper.mapProduct(product)));
     }
 
     public ProductDTO editProduct(ProductDTO product) {
-        if (repo
-                .findById(product.getId())
-                .isEmpty()) {
+        if (!productExists(product)) {
             throw new EntityNotFoundException(String.format("Couldn't find a product with the id %d", product.getId()));
         }
         return ProductMapper.mapProductWithDetails(repo
                 .save(ProductMapper.mapProduct(product)));
     }
 
-    public boolean checkIfProductExists(Long id){
-        return repo.existsById(id);
+    public void adjustAmountInStock(List<OrderItemDTO> receivedOrder) {
+        receivedOrder
+                .forEach(item -> {
+                    Product productToReceive = repo.getById(item.getProduct().getId());
+                    int newAmount = item.getQuantity() + productToReceive.getAmountInStock();
```

Comment:
This is the amount in stock in your shop, right? So on the suppliers' side the amount in stock would be reduced, but here it is added, right?

Reply:
I receive an order with orderItems on it. Each orderItem comes in a certain quantity. When I receive this order I take the amountInStock, add the quantity in that the respective product got delivered and save this newAmount in my stock.
I have no influence on the suppliers stock